### DISCUSSION TITLE
Conbee II installation and Udev rule help

### DISCUSSION THREAD

**Discussion by hermsdad** [Koenkk/zigbee2mqtt#6505] [Devices]
Trying to get my Conbee II set up. I am starting from scratch. Installed a fresh Raspberry Pi desktop (Debian) OS on to an old Mac Mini that I have. Completely deleted all Mac OS. 
I have hoobs up and running. The Conbee II has been upgraded and I have it working on a Raspberry 3+ with no problems. I just can't seem to get zigbee2mqtt to recognize it in this environment. I see in the FAQ's that I need to add a U-Dev rule. 

When I run ls I get:
lrwxrwxrwx 1 root root 13 Feb 28 23:09 usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2411659-if00 -> ../../ttyACM0

When I run usb-devices I get:
T:  Bus=03 Lev=02 Prnt=02 Port=01 Cnt=02 Dev#=  4 Spd=12  MxCh= 0
D:  Ver= 2.01 Cls=02(commc) Sub=00 Prot=00 MxPS=64 #Cfgs=  1
P:  Vendor=1cf1 ProdID=0030 Rev=01.00
S:  Manufacturer=dresden elektronik ingenieurtechnik GmbH
S:  Product=ConBee II
S:  SerialNumber=DE2411659
C:  #Ifs= 2 Cfg#= 1 Atr=80 MxPwr=100mA
I:  If#=0x0 Alt= 0 #EPs= 1 Cls=02(commc) Sub=02 Prot=01 Driver=cdc_acm
I:  If#=0x1 Alt= 0 #EPs= 2 Cls=0a(data ) Sub=00 Prot=00 Driver=cdc_acm

The FAQ says to add a rule: sudo nano /etc/udev/rules.d/99-usb-serial.rules
and add the following:
SUBSYSTEM=="tty", 
ATTRS{idVendor}=="0451", 
ATTRS{idProduct}=="16a8", 
ATTRS{serial}=="YOURSERIAL", 
SYMLINK="ttyUSB.CC2531-01", 
OWNER="pi"

so for my particular setup I am adding:
The FAQ says to add a rule: sudo nano /etc/udev/rules.d/99-usb-serial.rules
and add the following:
SUBSYSTEM=="tty", 
ATTRS{idVendor}=="1cf1", 
ATTRS{idProduct}=="0030", 
ATTRS{serial}=="DE2411659", 
SYMLINK="ttyUSB.CONBEE", 
OWNER="hoobs"

I am then adding:
serial: 
   port: /dev/ttyUSB.CONBEE
to /opt/zigbee2mqtt/data/configuration.yaml

When I do this it bricks my computer! Well not a total brick. It puts me in emergency mode and I have to delete the file. Any suggestions? I am using nano for edits.

**Comment by m0wlheld** [comment]
> Any suggestions?

Use /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2411659-if00 directly.

**Reply by hermsdad** [reply]
Forgot to mention, I'm a bit of a noob. Could you be more specific? Would I add that to "99-usb-serial.rules"? Of so where and how? Or would I not use "99-usb-serial.rules" and just add that to configuration.yaml?

Thanks for helping out.

**Reply by m0wlheld** [reply]
Assignment to generic USB device nodes (like /dev/ttyUSB0 oder /dev/ttyUSB1) depends on the order and count of USB devices pluged-in. It it therefore volatile and unstable. The udev rules are used to assign a USB device (based on it's identification) to a fix device node.

But - your USB coordinator device is also mapped to /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2411659-if00 and this is a stable path, too.

So, rollback your udev rule and use 

```
serial:
  port: /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE2411659-if00

```
in Zigbee2MQTT's /opt/zigbee2mqtt/data/configuration.yaml

**Reply by hermsdad** [reply]
Did that. However I still have permission issues. Which I believe was the whole reason to install the udev rule. 
When I run test -w /dev/ttyACM0 && echo success || echo failure as user hoobs I get failure. Then I run sudo chown hoobs /dev/ttyACM0 and if I run the test again I get success. As the FAQ points out I need to make sure permissions run at boot so I don't have to do that every time.
After doing chown if I run sudo npm start I get the following errors:
> node index.js

Zigbee2MQTT:info  2021-03-01 09:51:38: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2021-03-01.09-51-38' filename: log.txt
Zigbee2MQTT:info  2021-03-01 09:51:38: Starting Zigbee2MQTT version 1.17.1 (commit #6484fee)
Zigbee2MQTT:info  2021-03-01 09:51:38: Starting zigbee-herdsman (0.13.59)
Zigbee2MQTT:error 2021-03-01 09:51:58: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-03-01 09:51:58: Failed to start zigbee
Zigbee2MQTT:error 2021-03-01 09:51:58: Exiting...
Zigbee2MQTT:error 2021-03-01 09:51:58: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.<anonymous> (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:94:31)
    at Generator.throw (<anonymous>)
    at rejected (/opt/zigbee2mqtt/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:25:65)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.17.1 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.17.1 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-03-01T14_51_58_958Z-debug.log


Thanks for being patient with me.

**Reply by m0wlheld** [reply]
Check the group owner of /dev/ttyACM0, usually it’s „dialout“. You could add user hoobs to that group (or the user running Z2M).

**Reply by hermsdad** [reply]
well I started from scratch again. This time I didn't replace pi with hoobs. I just left everything as pi and ran the install scripts as written. It is running more smoothly. I need to test if I can add a device. I don't have a zigbee device with me at the moment. However it appears I do still have an error:
Zigbee2MQTT:info  2021-03-01 15:14:17: Logging to console and directory: '/opt/zigbee2mqtt/data/log/2021-03-01.15-14-17' filename: log.txt
Zigbee2MQTT:info  2021-03-01 15:14:18: Starting Zigbee2MQTT version 1.18.0 (commit #3fd6619)
Zigbee2MQTT:info  2021-03-01 15:14:18: Starting zigbee-herdsman (0.13.72)
CREATED DECONZ ADAPTER
Zigbee2MQTT:info  2021-03-01 15:14:18: zigbee-herdsman started
Zigbee2MQTT:info  2021-03-01 15:14:18: Coordinator firmware version: '{"meta":{"maintrel":0,"majorrel":38,"minorrel":74,"product":0,"revision":"0x264a0700","transportrev":0},"type":"ConBee2/RaspBee2"}'
Zigbee2MQTT:info  2021-03-01 15:14:21: Currently 0 devices are joined:
Zigbee2MQTT:warn  2021-03-01 15:14:21: `permit_join` set to  `true` in configuration.yaml.
Zigbee2MQTT:warn  2021-03-01 15:14:21: Allowing new devices to join.
Zigbee2MQTT:warn  2021-03-01 15:14:21: Set `permit_join` to `false` once you joined all devices.
Zigbee2MQTT:info  2021-03-01 15:14:21: Zigbee: allowing new devices to join.
Segmentation fault

When I had it working on my Raspberry 3+ I don't remember getting a segmentation fault. Like I said I'll check with a zigbee device to see if I can get it to join later this afternoon.

Thanks for being so patient with me. Don't know why I was trying to force making the install work with user hoobs. Not sure why I thought that was important. I'll update tonight if it works.

**Reply by m0wlheld** [reply]
Actually, I am facing the same error now with version 1.18.0. Was on 1.17.1 until now and it worked smoothly. Now with 1.18.0 it fails with the same error as yours:

```
Zigbee2MQTT:info  2021-03-01 22:42:12: Logging to console only'
Zigbee2MQTT:info  2021-03-01 22:42:12: Starting Zigbee2MQTT version 1.18.0 (commit #3fd6619)
Zigbee2MQTT:info  2021-03-01 22:42:12: Starting zigbee-herdsman (0.13.72)
Zigbee2MQTT:error 2021-03-01 22:42:33: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-03-01 22:42:33: Failed to start zigbee
Zigbee2MQTT:error 2021-03-01 22:42:33: Exiting...
Zigbee2MQTT:error 2021-03-01 22:42:33: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.<anonymous> (/app/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:94:31)
    at Generator.throw (<anonymous>)
    at rejected (/app/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:25:65)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! zigbee2mqtt@1.18.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the zigbee2mqtt@1.18.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-03-01T21_42_33_282Z-debug.log
```

Going back to 1.17.1 fixed it.

**Reply by hermsdad** [reply]
lol. How do I install 1.17.1?

**Reply by m0wlheld** [reply]
I use the Docker container image, so switching versions is easy.

**Reply by hermsdad** [reply]
I ended up having to install node 14.16.0 to fix the issue. I couldn't install it using the normal commands because for some reason the install script wouldn't let me get past 10.24.0 so I had to do it from source. Huge PITA but now I have everything working.
